index/scorch/snapshot_index.go

```diff
@@ -385,7 +385,7 @@ func (i *IndexSnapshot) DocumentVisitFieldTerms(id index.IndexInternalID,
 	for _, field := range fields {
 		if cachedFieldDocs, exists := ss.cachedDocs.cache[field]; exists {
 			if tlist, exists := cachedFieldDocs.docs[localDocNum]; exists {
-				terms := bytes.SplitN(tlist, ByteSeparator, -1)
+				terms := bytes.SplitN(tlist, []byte{TermSeparator}, -1)
```

Comment:
minor thing - not sure if []byte{TermSeparator} is doing an allocation, but for less garbage, might want to have it preallocated